packages/fuels-contract/src/script.rs

```diff
@@ -22,7 +35,213 @@ impl Script {
+            call.maturity,
+            script,
+            script_data,
+            inputs,
+            outputs,
+            vec![],
+        );
+        call.wallet.sign_transaction(&mut tx).await.unwrap();
+
+        Script::new(tx)
+    }
+
+    /// Given the necessary arguments, create a script that will be submitted to the node to call
+    /// the contract. The script is the actual opcodes used to call the contract, and the script
+    /// data is for instance the function selector. (script, script_data) is returned as a tuple
+    /// of hex-encoded value vectors
+    fn get_instructions(calls: Vec<&ContractCall>) -> (Vec<u8>, usize) {
+        let num_calls = calls.len();
+        let offset = Self::get_data_offset(num_calls);
+
+        let mut instructions = vec![];
+        for _ in 0..num_calls {
+            instructions.extend(Self::single_call_instructions(offset, 0));
+        }
+
+        instructions.extend(Opcode::RET(REG_ONE).to_bytes());
+
+        (instructions, offset)
+    }
+
+    /// The script data consists of the following items in the given order:
+    /// 1. Asset ID to be forwarded (AmountId::LEN)
+    /// 2. Amount to be forwarded (1 * WORD_SIZE)
+    /// 3. Contract ID (ContractID::LEN);
+    /// 4. Function selector (1 * WORD_SIZE);
+    /// 5. Calldata offset (optional) (1 * WORD_SIZE)
+    /// 6. Encoded arguments (optional) (variable length)
+    fn get_script_data(calls: Vec<&ContractCall>, offset: usize) -> Vec<u8> {
+        let mut script_data: Vec<u8> = vec![];
+
+        for call in calls {
+            script_data.extend(call.call_parameters.asset_id.to_vec());
+
+            let amount = call.call_parameters.amount as Word;
+            script_data.extend(amount.to_be_bytes());
+
+            script_data.extend(call.contract_id.as_ref());
+
+            script_data.extend(call.encoded_selector);
+
+            // If the method call takes custom inputs or has more than
+            // one argument, we need to calculate the `call_data_offset`,
+            // which points to where the data for the custom types start in the
+            // transaction. If it doesn't take any custom inputs, this isn't necessary.
+            if call.compute_calldata_offset {
+                // Offset of the script data relative to the call data
+                let call_data_offset =
+                    offset + AssetId::LEN + WORD_SIZE + ContractId::LEN + 2 * WORD_SIZE;
+                let call_data_offset = call_data_offset as Word;
+
+                script_data.extend(&call_data_offset.to_be_bytes());
+            }
+            script_data.extend(call.encoded_args.clone());
+        }
+
+        script_data
+    }
+
+    /// Returns the VM instructions for calling a contract method
+    /// We use the Opcode to call a contract: `CALL` pointing at the
+    /// following registers;
+    ///
+    /// 0x10 Script data offset
+    /// 0x11 Gas price
+    /// 0x12 Coin amount
```

Comment:
I think you forgot to add a previous comment  about an issue tracking this (0x11 Gas price TODO: https://github.com/FuelLabs/fuels-rs/issues/184)